File: homework/ba-nguyen/lesson-10/two-sum-less-than-k.ts
function twoSumLessThanK(numbs: number[], k: number): number {
  numbs.sort((a, b) => a - b);
  let lo = 0;
  let hi = numbs.length - 1;
  let res = -1;

  while (lo < hi) {
    const sum = numbs[lo] + numbs[hi];

    if (sum < k) {
      res = Math.max(res, sum);
      lo++;
    } else {
      hi--;
    }
  }

  return res;
}

File: homework/Long_Nguyen/lesson_10/TwoSumLessThanK.java
import java.util.Arrays;

/**
 * Created by Long Nguyen
 */
public class TwoSumLessThanK {
    public int twoSumLessThanK(int[] nums, int k) {
        Arrays.sort(nums);
        int first = 0;
        int second = nums.length - 1;
        int ans = -1;

        while (first < second) {
            int sum = nums[first] + nums[second];
            if (sum < k) {
                ans = Math.max(ans, sum);
                first++;
            } else {
                second--;
            }
        }

        return ans;
    }
}
